src/common/timing/start-time.js

```diff
@@ -6,31 +6,34 @@
 // Use various techniques to determine the time at which this page started and whether to capture navigation timing information
 
 import { mark } from './stopwatch'
-import { setOffset } from './now'
+import { setOffset, getLastTimestamp } from './now'
 import { globalScope } from '../util/global-scope'
 
 export let navCookie = true
 
-// findStartTime() *cli - comment out to add agentId context which is n/a at import time
-
+/**
+ *
+ * @param {string} agentId - agentIdentifier tied to agent instance
+ * @returns {number} - Integer timestamp
+ */
 export function findStartTime (agentId) {
-  // findStartCookie is used for FF7/8 & browsers which do not support window.performance.timing.navigationStart
-  var starttime = findStartWebTiming() // || findStartCookie() -- now redundant *cli oct'22, TO DO: slated for removal
+  var starttime = findStartWebTiming()
   if (!starttime) return
 
   mark(agentId, 'starttime', starttime)
   setOffset(starttime)
+  return starttime
 }
 
 // Find the start time from the Web Timing 'performance' object.
-// The "timeOrigin" property is the new standard timestamp property shared across main frame and workers, but is not supported in some early Safari browsers (safari<15)
+// The "timeOrigin" property is the new standard timestamp property shared across main frame and workers, but is not supported in some early Safari browsers (safari<15) + IE
 // navigationStart is deprecated, but can still be used as a fallback
 // http://test.w3.org/webperf/specs/NavigationTiming/
 // http://blog.chromium.org/2010/07/do-you-know-how-slow-your-web-page-is.html
 function findStartWebTiming () {
   // note that we don't need to use a cookie to record navigation start time
   navCookie = false
-  const timestamp = globalScope?.performance?.timeOrigin || globalScope?.performance?.timing?.navigationStart
-  if (!timestamp) return
-  return Math.round(timestamp)
+  const timestamp = globalScope?.performance?.timeOrigin || globalScope?.performance?.timing?.navigationStart || getLastTimestamp()
```

Comment:
May I suggest doing this inside `now.js` file instead? I'm touching the same area with offset, and also am removing this function in the cwv branch

Reply:
Aside: `findStartTime()` is actually doing the same job as now.js with `setOffset` so it's also good to just trim one of them or use now.js inside findStartTime